File: c-sharp-runtime/src/ClosedRange.cs
using System;
using System.Runtime.InteropServices;
using static Cricut.FishyJoesRuntime.Utilities;

namespace Cricut.FishyJoesRuntime {
    // TODO: Docs
    // TODO: Easy conversion between System.Range and this type
    public record ClosedRange<T> {
        public T start;
        public T endInclusive;

        public ClosedRange(T start, T endInclusive) {
            this.start = start;
            this.endInclusive = endInclusive;
        }
    }
}

File: kotlin-runtime/src/main/kotlin/com/cricut/fishyjoes/runtime/ClosedRange.kt
package com.cricut.fishyjoes.runtime

import kotlin.ranges.ClosedRange

data class ClosedRange<T: Comparable<T>>(
    override var start: T,
    override var endInclusive: T
): ClosedRange<T> {
    constructor(range: kotlin.ranges.ClosedRange<T>): this(range.start, range.endInclusive)
}
